src/Controls/src/Core/SetterSpecificity.cs

```diff
@@ -0,0 +1,96 @@
+﻿#nullable disable
+using System;
+
+namespace Microsoft.Maui.Controls
+{
+	//NOTE: IDEA: review this: merge FROM into a single int (vsm, manual, dynamicR, binding), and CSS into another
+	internal readonly struct SetterSpecificity : IComparable<SetterSpecificity>
+	{
+		public static readonly SetterSpecificity DefaultValue = new(-1, 0, 0, 0, -1, 0, 0, 0);
+		public static readonly SetterSpecificity VisualStateSetter = new SetterSpecificity(1, 0, 0, 0, 0, 0, 0, 0);
+		public static readonly SetterSpecificity FromBinding = new SetterSpecificity(0, 0, 0, 1, 0, 0, 0, 0);
+
+		public static readonly SetterSpecificity ManualValueSetter = new SetterSpecificity(0, 100, 0, 0, 0, 0, 0, 0);
+		public static readonly SetterSpecificity Trigger = new SetterSpecificity(0, 101, 0, 0, 0, 0, 0, 0);
+
+		public static readonly SetterSpecificity DynamicResourceSetter = new SetterSpecificity(0, 0, 1, 0, 0, 0, 0, 0);
+
+		//handler always apply, but are removed when anything else comes in. see SetValueActual
+		public static readonly SetterSpecificity FromHandler = new SetterSpecificity(1000, 0, 0, 0, 0, 0, 0, 0);
+
+		//100-n: direct VSM (not from Style), n = max(99, distance between the RD and the target)
+		public int Vsm { get; }
+
+		//1: SetValue, SetBinding
+		public int Manual { get; }
+
+		//1: DynamicResource
+		public int DynamicResource { get; }
+
+		//1: SetBinding
+		public int Binding { get; }
+
+		//XAML Style specificity
+		//100-n: implicit style, n = max(99, distance between the RD and the target)
+		//200-n: RD Style, n = max(99, distance between the RD and the target)
+		//200: local style, inline css,
+		//300-n: VSM, n = max(99, distance between the RD and the target)
+		//300: !important (not implemented)
+		public int Style { get; }
```

Comment:
The number ranges here feel like "magic numbers", should we declare a `const int` for some of these?

When some of the unit tests do things like: `new SetterSpecificity(100, 0, 0, 0)` this seems like it would be clearer:

```csharp
new SetterSpecificity(SetterSpecificity.Implicit, 0, 0, 0)
```

Reply:
I'm not yet 100% sure about the internal implementation of specificity, see some notes left around. those numbers were a kind of a spec I wrote prior to coding this. any suggestion welcome (reduce the # of fields, use flags, magic numbers, etc...)

Reply:
I think if someone would commonly use the numbers 100, 200, 300, we could just start with a `const int` for those for now.